### DISCUSSION TITLE
Request twice in one server using broadcast2

### DISCUSSION THREAD

**Discussion by LuSingTea** [apache/dubbo#13876] [Question]
- [ ] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.

## Ask your question here
抱歉因为代码在公司内网，所以这里只能口头描述

**Comment by LuSingTea** [comment]
抱歉因为代码在公司内网，所以这里只能口头描述
dubbo版本为2.7.23
dubbo服务部署在k8s集群上通过Service NodePort对外暴露服务，消费者通过broadcast2模式调用时，会重复往其中一个提供者发送请求，请问各位有遇到过吗？

**Comment by AlbumenJ** [comment]
> 抱歉因为代码在公司内网，所以这里只能口头描述 dubbo版本为2.7.23 dubbo服务部署在k8s集群上通过Service NodePort对外暴露服务，消费者通过broadcast2模式调用时，会重复往其中一个提供者发送请求，请问各位有遇到过吗？

这里的 Service NodePort 怎么和 dubbo 关联上的

**Comment by LuSingTea** [comment]
```shell
spec:
  type: NodePort
  ports:
     -name: dubbo
       port: 30888
       targetPort: 30888
       nodePort: 30888
       protocol: tcp
    selector:
      app: xxxx
```
yaml配置文件如上，dubbo暴露端口为30888，有时候请求是正常的，有时候是不正常的

**Comment by LuSingTea** [comment]
补充说明：
（1）k8s的节点有3个
（2）如果部署了4个pod会出现往其中一个pod调用3次，另一个pod调用1次的情况

**Reply by yellowsand1** [reply]
我麻了 我先是遇到broadcast2的invoke里面如果方法返回void会报NPE  现在搜了一下issue  感觉不少broadcast的问题啊  最后是如何解决的呢 目前我们的功能还在开发中  不知道到时候会不会也遇到这样的情况
